### ISSUE TITLE
feat(bert): port BERT and XLM-RoBERTa encoders for /v1/embeddings with sequence-classification heads

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1321]
## Summary

mlxcel has no encoder-only transformer. BERT (`model_type: bert`) and XLM-RoBERTa (`model_type: xlm-roberta`) share one post-norm encoder block and differ only in position-id construction, `type_vocab_size`, `layer_norm_eps`, `pad_token_id` and the weight-key prefix, so they are ported together as `src/models/bert.rs` with a `BertVariant` switch. This delivers the most common sentence-embedding checkpoints (all-MiniLM, multilingual-e5, bge-m3) through `/v1/embeddings`, and the `BertForSequenceClassification` / `XLMRobertaForSequenceClassification` heads that #1356 needs.

## Current behavior

- `src/models/detection.rs:124-199` has no `"bert"` or `"xlm-roberta"` arm; `get_model_type` returns `Unsupported model type: bert`.
- No file under `src/models/` implements absolute position embeddings plus post-LayerNorm blocks; the nearest code is the SigLIP vision encoder `src/vision/encoders/siglip.rs:147-260` (pre-norm, no token-type embeddings, no mask) and `mlxcel_core::layers::LayerNorm` (`src/lib/mlxcel-core/src/layers.rs:945`, weight + optional bias).
- #1353 adds `ModelType::Bert` and `ModelType::XlmRoberta` as registered-but-unloadable variants; this issue supplies the loader and the model.

### Config (`config.json`)

| key | BERT default | XLM-R default | note |
|---|---|---|---|
| `vocab_size` | required | required | |
| `hidden_size` | required | required | `D` |
| `num_hidden_layers` | required | required | |
| `num_attention_heads` | required | required | `head_dim = D / heads` |
| `intermediate_size` | required | required | |
| `max_position_embeddings` | required | required | position table rows (XLM-R stores `max_len + 2`, e.g. 8194) |
| `type_vocab_size` | 2 | 1 | |
| `layer_norm_eps` | 1e-12 | 1e-5 | |
| `pad_token_id` | 0 | 1 | XLM-R `padding_idx` |
| `hidden_act` | `gelu` | `gelu` | exact erf GELU, `mlxcel_core::gelu` |
| `num_labels` / `id2label` | 1 | 1 | classification head width; `num_labels = len(id2label)` when `num_labels` absent |

Detection: `model_type == "bert"` -> `ModelType::Bert`, `"xlm-roberta"` -> `ModelType::XlmRoberta`; `architectures[0]` ending in `ForSequenceClassification` selects the classifier head (consumed by #1356), anything else (`BertModel`, `BertForMaskedLM`, `XLMRobertaModel`, `XLMRobertaForMaskedLM`) is the embedder.

### Forward

Inputs: `input_ids [B, L]`, `attention_mask [B, L]` (1 = real), `token_type_ids [B, L]` (zeros when absent).

```
BERT:   position_ids[b, i] = i
XLM-R:  m = (input_ids != pad_token_id); position_ids = cumsum(m, axis=1) * m + pad_token_id
h = word_embeddings[input_ids] + position_embeddings[position_ids] + token_type_embeddings[token_type_ids]
h = LayerNorm(h)                                        # embeddings.LayerNorm
mask = create_bidirectional_padding_mask(attention_mask)  # [B,1,1,L], f32 0/-inf
for layer in 0..num_hidden_layers:
    q = query(h); k = key(h); v = value(h)                # all with bias
    reshape to [B, heads, L, head_dim]; attn = sdpa(q, k, v, scale = head_dim^-0.5, mask)
    a = LayerNorm(attention.output.dense(attn) + h)       # attention.output.LayerNorm
    f = output.dense(gelu(intermediate.dense(a)))
    h = LayerNorm(f + a)                                  # output.LayerNorm
last_hidden_state = h                                      # [B, L, D]
embedding = normalize_l2(pool(h, attention_mask, mode))    # mode from 1_Pooling, default mean
```

Classifier head (sequence-classification checkpoints):
```
BERT:   logits = classifier(tanh(pooler.dense(h[:, 0, :])))               # classifier: [num_labels, D]
XLM-R:  logits = classifier.out_proj(tanh(classifier.dense(h[:, 0, :])))  # out_proj: [num_labels, D]
```

### Weight keys and sanitize

Checkpoint keys (verified on `sentence-transformers/all-MiniLM-L6-v2` and `BAAI/bge-reranker-v2-m3`):

```
[bert.|roberta.]embeddings.word_embeddings.weight        [vocab, D]
[bert.|roberta.]embeddings.position_embeddings.weight    [max_pos, D]
[bert.|roberta.]embeddings.token_type_embeddings.weight  [type_vocab, D]
[bert.|roberta.]embeddings.LayerNorm.{weight,bias}
[bert.|roberta.]embeddings.position_ids                  (buffer, drop)
[bert.|roberta.]encoder.layer.{i}.attention.self.{query,key,value}.{weight,bias}
[bert.|roberta.]encoder.layer.{i}.attention.output.dense.{weight,bias}
[bert.|roberta.]encoder.layer.{i}.attention.output.LayerNorm.{weight,bias}
[bert.|roberta.]encoder.layer.{i}.intermediate.dense.{weight,bias}
[bert.|roberta.]encoder.layer.{i}.output.dense.{weight,bias}
[bert.|roberta.]encoder.layer.{i}.output.LayerNorm.{weight,bias}
pooler.dense.{weight,bias}            (BERT; kept only for the classifier)
classifier.{weight,bias}              (BERT classifier)
classifier.dense.*, classifier.out_proj.*  (XLM-R classifier)
cls.*  / lm_head.*                    (MLM heads, drop)
```

Sanitize rule (`src/models/bert.rs::sanitize`): strip a leading `bert.` or `roberta.`; drop any key containing `position_ids`, starting with `cls.` or `lm_head.`; for the embedder also drop `pooler.`. Idempotent (a second pass is a no-op), matching the convention in `src/models/sanitize.rs`.

### Limits and pooling defaults

- Default pooling `mean`; `1_Pooling/config.json` overrides (all-MiniLM: `pooling_mode_mean_tokens: true`; e5: mean; bge-m3: `pooling_mode_cls_token: true`).
- `max_length` = min(`sentence_bert_config.json` `max_seq_length` (256 for all-MiniLM), `tokenizer_config` `model_max_length`, `config.max_position_embeddings`, 8192). Absolute positions make `max_position_embeddings` a hard limit: XLM-R additionally subtracts `pad_token_id + 1` (positions start at 2), so bge-m3 caps at 8192 real tokens.
- Tokenization with `add_special_tokens = true` (`[CLS] ... [SEP]` / `<s> ... </s>` come from the tokenizer's post-processor); `needs_token_type_ids() == true` for BERT (zeros for single texts), false for XLM-R.
- Non-quantized f32 checkpoints are run in f32 (do not convert to f16: all-MiniLM ships f32 and the 1e-12 LayerNorm eps underflows in f16); bf16 checkpoints follow the text rule (bf16 -> f16). Quantized checkpoints with `config.quantization` load every `*.dense`, `query/key/value` and the word embedding through `UnifiedLinear` / `UnifiedEmbedding` (per-tensor `.scales` presence decides).

## Implementation plan

1. `src/models/bert.rs` (+ `bert_tests.rs`): `BertArgs` (serde, the table above, `#[serde(default)]` for defaults, `#[serde(alias = "xlm-roberta")]` not needed since `model_type` is read by detection), `BertVariant { Bert, XlmRoberta }`, `BertEmbeddings { word, position, token_type: UnifiedEmbedding, layer_norm: LayerNorm, variant, pad_token_id }`, `BertLayer { query, key, value, attn_out, intermediate, output: UnifiedLinear, attn_norm, out_norm: LayerNorm }`, `BertEncoder { embeddings, layers, num_heads, scale }` with `fn encode(&self, input_ids, attention_mask, token_type_ids) -> [B, L, D]` using `mlxcel_core::layers::attention` with the explicit mask, `mlxcel_core::gelu`, `mlxcel_core::cumsum` / `not_equal` / `multiply` / `add_scalar` for XLM-R positions. Constructors `from_weights(weights, args, variant, group_size, bits)`.
2. `BertEmbeddingModel { encoder, pooling: PoolingMode, normalize: bool }` implementing `crate::embeddings::EmbeddingModel` (`embed`, `default_pooling = Mean`, `embedding_dim = hidden_size`, `needs_token_type_ids = variant == Bert`).
3. `BertSequenceClassifier { encoder, head: BertClassifierHead | RobertaClassifierHead }` with `fn logits(&self, input_ids, attention_mask, token_type_ids) -> [B, num_labels]`, exposed for #1356 (no HTTP wiring here).
4. `src/models/detection.rs`: `"bert" => Bert`, `"xlm-roberta" | "xlm_roberta" => XlmRoberta`; `src/model_metadata.rs` entries with `kind: Embedding`; `src/embeddings/loader.rs` arms calling `BertEmbeddingModel::load(dir)` which reads `config.json`, loads weights with `load_weights_from_dir_with_subfolders`, applies `sanitize`, converts dtype per the rule above, resolves pooling through `PoolingConfig::read`, and derives limits through `EmbeddingLimits::derive(..., is_absolute_position = true)`.
5. `docs/supported-models.md` embedding table rows for BERT and XLM-RoBERTa; `docs/embeddings.md` note on e5's `query: ` / `passage: ` prefixes and bge-m3's CLS pooling.

## Validation

(a) Unit tests in `src/models/bert_tests.rs`:
- `sanitize_strips_prefix_and_drops_mlm_and_buffers` (keys from the list above, both prefixes, idempotent).
- `xlm_roberta_position_ids_start_after_padding_idx` (`[5, 6, 1, 1]` with pad 1 gives `[2, 3, 1, 1]`).
- `bert_forward_shapes_and_finite` on a 2-layer random-weight config, `[2, 5]` input with one padded row.
- `padding_row_does_not_change_real_row_embedding`: embed `["a b c"]` alone and `["a b c", "a b c d e f"]` batched, cosine of the first row between the two runs `> 1 - 1e-5`.
- `classifier_head_shapes` for both variants with `num_labels = 1`.

(b) Real checkpoints (all under 2.5 GB):
```bash
./target/release/mlxcel download sentence-transformers/all-MiniLM-L6-v2
./target/release/mlxcel embed -m models/all-MiniLM-L6-v2 \
  -p "The weather is lovely today." -p "It's so sunny outside!" -p "He drove to the stadium."
./target/release/mlxcel-server -m models/all-MiniLM-L6-v2 --port 8080
curl -s localhost:8080/v1/embeddings -H 'Content-Type: application/json' \
  -d '{"input": ["The weather is lovely today.", "It'"'"'s so sunny outside!", "He drove to the stadium."]}'
```
Acceptance: cosine(0, 1) = 0.666 and cosine(0, 2) = 0.105 within 1e-2 (the values published in the sentence-transformers quickstart for this checkpoint), each vector has unit norm within 1e-5, and the batched result equals the three single-input results within 1e-4.

```bash
./target/release/mlxcel download BAAI/bge-m3
./target/release/mlxcel embed -m models/bge-m3 -p "What is BGE M3?" -p "Defination of BM25" \
  -p "BGE M3 is an embedding model supporting dense retrieval, lexical matching and multi-vector interaction." \
  -p "BM25 is a bag-of-words retrieval function that ranks a set of documents based on the query terms appearing in each document"
```
Acceptance (XLM-RoBERTa path, CLS pooling): similarity matrix rows for the two queries against the two passages equal `[[0.6265, 0.3477], [0.3499, 0.678]]` within 1.5e-2 (bge-m3 model card values), and a 1200-token input embeds without error (exercises the shifted position ids beyond 512).

`intfloat/multilingual-e5-small` (BERT, 12 layers, 250k vocab) is the third checkpoint: `query: ` / `passage: ` prefixed pairs rank the matching passage first.

## Acceptance criteria

- [x] `src/models/bert.rs` loads BERT and XLM-RoBERTa embedders and their `ForSequenceClassification` variants from the key layouts above (the two heads live in `src/models/bert_heads.rs`; the trunk stays under the 500-line guideline)
- [x] XLM-RoBERTa position ids follow `cumsum(mask) * mask + pad_token_id`
- [x] all-MiniLM-L6-v2 and bge-m3 pass the numeric gates above through both `mlxcel embed` and `POST /v1/embeddings` (bge-m3 ran against the `seansitter/bge-m3-safetensors` mirror, because the BAAI repo publishes only `pytorch_model.bin` and the downloader takes safetensors)
- [x] docs/supported-models.md updated, plus a `Family notes` section in docs/embeddings.md covering the e5 `query: ` / `passage: ` prefixes and bge-m3's CLS pooling
- [x] detection table in src/models/detection.rs updated with a test (already landed with #1353: `embedding_variant_for_model_type` has the `bert` and `xlm-roberta` arms and `src/models/detection_tests.rs` covers both)
- [x] tests pass in the CUDA form of that command on this Linux box: `cargo test --profile test-fast --features cuda --lib models::bert` (27 passed, three repeats) and `--lib embeddings::` (63 passed). The `metal,accelerate` feature pair cannot be built here; the epic's Apple Silicon pass covers it.
- [x] `cargo clippy --profile test-fast --features cuda --lib --bins --tests -- -D warnings` and `cargo fmt --all -- --check` pass

## Out of scope

- `/v1/rerank` wiring of the classifier head (#1356).
- Sparse or ColBERT outputs of bge-m3 (dense only).
- DistilBERT, RoBERTa (non-XLM), DeBERTa.

## Dependencies

Blocked by #1353

- Part of epic #1348
